SampleScripts/PowerManagement-WorkloadDomain.ps1

```diff
@@ -254,6 +254,42 @@ Try {
         [Array]$vcfvms = @()
         [Array]$vcfvms += ($vcServer.fqdn).Split(".")[0]
 
+        # Gather VxRail Manager Details for the VI Workload Domain, if exists
+        if ($PsBoundParameters.ContainsKey("shutdown")) {
+            $vxRailCred = (Get-VCFCredential | Where-Object { $_.resource.resourceType -eq "VXRAIL_MANAGER" -and $_.resource.domainName -eq ($workloadDomain.name) -and $_.username -eq "root" })
+            if ($vxRailCred -ne $null)
+            {
+			    # Connecting to VC to get VxRail VM name
+			    if ($DefaultVIServers) {
+                        Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue | Out-Null
+                }
+                if ((Test-NetConnection -ComputerName $vcServer.fqdn -Port 443 ).TcpTestSucceeded) {
+                    Write-PowerManagementLogMessage -Type INFO -Message "Connecting to '$($vcServer.fqdn)' ..."
+                    Connect-VIServer -Server $vcServer.fqdn -Protocol https -User $vcUser -Password $vcPass -ErrorVariable $vcConnectError | Out-Null
+                    if ($DefaultVIServer.Name -eq $vcServer.fqdn) {
+					    $vxrailVMObject = Get-VM | where-object {$_.Guest.Hostname -match $vxRailCred.resource.resourceName -Or $_.Guest.Hostname -match ($vxRailCred.resource.resourceName.Split("."))[0]}
+					    if ($vxrailVMObject) {
+					    	$vxRailVmName = $vxrailVMObject.Name
+					    } else {
+					    	Write-PowerManagementLogMessage -Type ERROR -Message "VxRail($($vxRailCred.resource.resourceName)) Virtual Machine object cannot be located within VC Server ($($vcServer.fqdn))" -Colour Red
+					    }
+				    }
+			    }
+			
+                $vxRailDetails = New-Object -TypeName PSCustomObject
+                $vxRailDetails | Add-Member -Type NoteProperty -Name fqdn -Value $vxRailCred.resource.resourceName
+			    $vxRailDetails | Add-Member -Type NoteProperty -Name vmName -Value $vxRailVmName
+                $vxRailDetails | Add-Member -Type NoteProperty -Name username -Value $vxRailCred.username
+                $vxRailDetails | Add-Member -Type NoteProperty -Name password -Value $vxRailCred.password
+                [Array]$vcfvms += ($vxRailDetails.vmName)
+                Write-PowerManagementLogMessage -Type INFO -Message "VxRail Manager($vxRailVmName) found within VC Server ($($vcServer.fqdn))" -Colour Yellow
+            }
```

Comment:
`} else {`

Reply:
addressed in commit f8c5a69ef84a1a40675a91879ad0a67701fba108